index.d.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * All exports from TS source files (where the implementation is actualy done in TS).
+ */
+export * from './types/type_exports';
```

Comment:
There is a top level `typings` folder too, I think `types` is new in this PR? I think this will end up being confusing. Maybe they can be merged? 

https://github.com/elastic/kibana/tree/master/typings

Reply:
The `types` directory is built from `yarn build:types` which gets run on `yarn kbn bootstrap`. In master it goes to the `target/types` directory but in this PR I changed it to `types`.

The reason I changed it to not be `target` is that the build wipes out that directory before building and I was previously trying to import from it which was failing Typescript build for x-pack. Let me confirm, but I think after some of the other changes I made, I can put this back the way it was before.

Reply:
Confirmed, I'll put this back to pulling from the `target` directory to reduce confusion. Thanks for catching this!